packages/broker/src/ConfigWizard.ts

```diff
@@ -82,200 +84,170 @@ const DefaultConfig: Config = {
-            name:'generateOrImportEthereumPrivateKey',
-            message: 'Do you want to generate a new Ethereum private key or import an existing one?',
-            choices: ['Generate', 'Import'],
-            filter: (choice: string) => {
-                return choice.toLowerCase()
+const pluginPrompts: Array<inquirer.Question | inquirer.ListQuestion | inquirer.CheckboxQuestion> = []
+Object.keys(PLUGIN_TEMPLATES).map((pluginName) => {
+    const plugin = PLUGIN_TEMPLATES[pluginName]
+    pluginPrompts.push({
+        type: 'input',
+        name: `${pluginName}Port`,
+        message: `Select a port for the ${pluginName} Plugin [Enter for default: ${plugin.port}]`,
+        when: (answers: inquirer.Answers) => {
+            return answers.selectPlugins.includes(pluginName)
+        },
+        validate: (input: string, answers: inquirer.Answers): string | boolean => {
+            const portNumber = parseInt(input || answers[`${pluginName}Port`])
+            if (!Number.isInteger(portNumber)) {
+                return `Non-numeric value provided`
             }
-        })
 
-        let wallet: Wallet
-        if (generateOrImport === 'generate') {
-            wallet = Wallet.createRandom()
-            this.config.ethereumPrivateKey = wallet.privateKey
-            logger.info('This is your generated private key. Please store it in a secure location')
-            logger.alert(this.config.ethereumPrivateKey)
-        } else if (generateOrImport === 'import') {
-            const privateKey = await this.inquirerSinglePrompt({
-                type: 'input',
-                name: 'privateKey',
-                message: "'Please provide the private key to import'",
-            })
-
-            try {
-                wallet = new Wallet(privateKey)
-                this.config.ethereumPrivateKey = wallet.privateKey
-            } catch (privateKeyError) {
-                throw new Error(`Invalid privateKey provided for import: ${privateKey}`)
+            if (portNumber < MIN_PORT_VALUE || portNumber > MAX_PORT_VALUE) {
+                return `Out of range port ${portNumber} provided (valid range ${MIN_PORT_VALUE}-${MAX_PORT_VALUE})`
             }
 
-        } else {
-            throw new Error(`Invalid option ${generateOrImport} provided`)
+            return true
+        },
+        default: plugin.port
+    })
+})
+
+prompts = prompts.concat(pluginSelectorPrompt).concat(pluginPrompts)
+
+export const getConfigFromAnswers = (answers: any): Config => {
+    const config = { ... DEFAULT_CONFIG, plugins: { ... DEFAULT_CONFIG.plugins } }
+
+    const pluginNames = Object.keys(PLUGIN_TEMPLATES)
+    pluginNames.forEach((pluginName) => {
+        const template = PLUGIN_TEMPLATES[pluginName]
+        if (answers.selectPlugins && answers.selectPlugins.includes(pluginName)){
+            // the publishHttp plugin is special, it needs to be added to the config after the other plugins
+            if (pluginName === 'publishHttp') {
```

Comment:
If we refer to this value twice, it should be a constant (maybe all plugin names could be)

Reply:
with the latest commit it is not repeated anymore, seems quite redundant to implement all plugin names as constants for single usage

Reply:
It is still at line 126 and line 173
```
publishHttp: { port: DEFAULT_HTTP_PORT }
if (pluginName === 'publishHttp') {
```